index.js

```diff
@@ -1,26 +1,91 @@
 require("dotenv").config();
 const express = require("express");
+const bcrypt = require("bcryptjs");
 
 const db = require("./data/knexConfig");
 
 const port = process.env.PORT || 4000;
 
 const server = express();
 
+server.use(express.json());
+
 server.get("/", (req, res) => {
   res.send('welcome to the "/"');
 });
 
-server.get("/api/users", async (req, res) => {
+server.post("/api/register", async (req, res) => {
+  console.log("req.body: ", req.body);
+  const user = req.body;
+  if (user.username && user.password) {
+    const hash = bcrypt.hashSync(user.password, 4);
+    user.password = hash;
+    try {
+      console.log(user);
+      const id = await db("users").insert(user);
+      try {
+        const user = await db("users")
+          .where({ id: id })
```

Comment:
```suggestion
          .where({ id })
``` Since the variables are the same, you can just write it like this